### PR TITLE
add shopping cart example tests (for #1481)

### PR DESCRIPTION
**Pull request by fshowalter**
Adds component and reducer tests for the shopping cart example. Component tests use enzyme.

@gaearon please let me know if you'd like a different format for the reducer tests. I took a best-guess shot based on the counter example, but I'm happy to change them.

### CODE REVIEW

File: examples/shopping-cart/package.json [hunk 1]
```diff
@@ -28,9 +30,14 @@
     "babel-preset-es2015": "^6.3.13",
     "babel-preset-react": "^6.3.13",
     "babel-preset-react-hmre": "^1.1.1",
+    "cross-env": "^1.0.7",
+    "enzyme": "^2.0.0",
     "express": "^4.13.3",
     "json-loader": "^0.5.3",
+    "react-addons-test-utils": "^0.14.7",
```
Comment (gaearon) [inline examples/shopping-cart/package.json]: Do we need these? Does `enzyme` peerDep on them or is it here by accident?

**Comment by gaearon** [comment]
This looks great, thank you.

File: examples/shopping-cart/test/reducers/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import expect from 'expect'
+import { getTotal, getCartProducts } from '../../reducers'
+
+describe('reducers', () => {
```
Comment (gaearon) [inline examples/shopping-cart/test/reducers/index.spec.js]: Maybe worth changing to `selectors`? They are grouped with reducers in the code for convenience but we should probably label them differently.

File: examples/shopping-cart/package.json (hunk 1 above)
Comment (fshowalter) [inline examples/shopping-cart/package.json]: Needed if you're using React 0.14. Enzyme works with 0.13 or 0.14 but only requires `react-addons-test-utils` if you're using 0.14. Thus, they can't peerDep it, but they do document it. 

`react-dom` is required in 0.14 too, but we're using shallow rendering so we don't need it.

File: examples/shopping-cart/test/reducers/index.spec.js (hunk 2 above)
Comment (fshowalter) [inline examples/shopping-cart/test/reducers/index.spec.js]: Good call. Updated.

**Comment by timdorr** [comment]
This is great! Very readable and comprehendible tests!

**Pull request merged**